### ISSUE TITLE
Global augmentations don't expose all merged members

### ISSUE THREAD

**Issue by berwyn** [Microsoft/TypeScript#9279]
**TypeScript Version:** 

1.8.10

**Code**

```ts
// jquery.d.ts
interface JQuery {
    on(events: string, handler: (eventObject: JQueryEventObject, ...args: any[]) => any): JQuery;
}

// widget.ts
declare global {
    interface JQuery {
        testWidget(): JQuery;
        testWidget(options: TestWidgetOptions): JQuery;
    }
}

export interface TestWidgetOptions {
    first?: number;
    second?: number;
}

$.widget('testWidget', { /* elided */ });

// test.ts
$('foo').testWidget().on('someEvent', evt => { /* elided */ });

```
**Expected behavior:**

The function declared in the augmented module should return the properly augmented variable, not just it's local implementation.

In this case, the returned variable should have access to the complete set of JQuery instance members.

**Actual behavior:** 

The type checker errors if any instance member is used other than those declared in the augmentation. In this case, JQuery objects should have the `on(events: string, handler: (eventObject: JQueryEventObject, ...args: any[]) => any): JQuery;` function, but it's not visible to JQuery objects returned from global augmentations.

This can be circumvented by casting the returned value to its same interface (`$('foo').testWidget as JQuery`), which leads to much uglier code and developer confusion.

This can be recreated with ambient type definitions installed via the `typings` tool and with any given module augmentation where a function returns a `JQuery` object.

**Comment by vladima** [comment]
I can repro this in release 1.8 however it seems to be fixed in master. Can you try `typescript@next` on npm to see if this issue still exist?

**Comment by berwyn** [comment]
Yup, `typescript@next` resolves the issue! We should be able to close this as resolved in 1.9 then.
